syft/pkg/package.go

```diff
@@ -33,6 +33,14 @@ func (p *Package) OverrideID(id artifact.ID) {
 }
 
 func (p *Package) SetID() {
+	if p.Name == "" {
```

Comment:
Another approach is to create a  package constructor and refactor it all over the code base.